src/iceberg/catalog/memory_catalog.h

```diff
@@ -0,0 +1,228 @@
+   */
+  bool DeleteNamespace(const Namespace& namespace_ident);
+
+  /**
+   * @brief Retrieves the properties of the specified namespace.
+   * @param namespace_ident The namespace whose properties to retrieve.
+   * @return An optional containing the properties map if the namespace exists;
+   *         std::nullopt otherwise.
+   */
+  std::optional<std::unordered_map<std::string, std::string>> GetProperties(
+      const Namespace& namespace_ident) const;
+
+  /**
+   * @brief Replaces all properties of the given namespace.
+   * @param namespace_ident The namespace whose properties will be replaced.
+   * @param properties The new properties map.
+   * @return True if the namespace exists and properties were replaced; false otherwise.
+   */
+  bool ReplaceProperties(const Namespace& namespace_ident,
+                         const std::unordered_map<std::string, std::string>& properties);
+
+  /**
+   * @brief Lists all table names under the specified namespace.
+   * @param namespace_ident The namespace from which to list tables.
+   * @return A vector of table names.
+   */
+  std::vector<std::string> ListTables(const Namespace& namespace_ident) const;
+
+  /**
+   * @brief Registers a table in the given namespace with a metadata location.
+   * @param table_ident The fully qualified identifier of the table.
+   * @param metadata_location The path to the table's metadata.
+   * @return True if the table was registered successfully; false otherwise.
+   */
+  bool RegisterTable(TableIdentifier const& table_ident,
+                     const std::string& metadata_location);
+
+  /**
+   * @brief Unregisters a table from the specified namespace.
+   * @param table_ident The identifier of the table to unregister.
+   * @return True if the table existed and was removed; false otherwise.
+   */
+  bool UnregisterTable(TableIdentifier const& table_ident);
+
+  /**
+   * @brief Checks if a table exists in the specified namespace.
+   * @param table_ident The identifier of the table to check.
+   * @return True if the table exists; false otherwise.
+   */
+  bool TableExists(TableIdentifier const& table_ident) const;
+
+  /**
+   * @brief Gets the metadata location for the specified table.
+   * @param table_ident The identifier of the table.
+   * @return An optional string containing the metadata location if the table exists;
+   *         std::nullopt otherwise.
+   */
+  std::optional<std::string> GetTableMetadataLocation(
+      TableIdentifier const& table_ident) const;
+
+ private:
+  /**
+   * @brief Helper function to retrieve the container node for a given namespace.
+   * @param root The root of the tree to search.
+   * @param namespace_ident The namespace path to traverse.
+   * @return A pointer to the corresponding NamespaceContainer if it exists; nullptr
+   * otherwise.
+   */
+  static NamespaceContainer* GetNamespaceContainer(NamespaceContainer* root,
```

Comment:
Functions like these can be moved to memory_catalog.cc in an anonymous namespace.